src/Ch9/Ch9.Shared/App.xaml.cs

```diff
@@ -179,6 +179,45 @@ private void ConfigureOrientation()
 			{
 				DisplayInformation.AutoRotationPreferences = DisplayOrientations.Portrait;
 			}
+
+			void OnOrientationChanged(DisplayInformation displayInformation, object args)
+			{
+				ShowViewModel showViewModel = null;
+
+				if ((_rootFrame.Content as FrameworkElement)?.DataContext is ShowPageViewModel showPage)
+				{
+					showViewModel = showPage.Show;
+				}
+				else if ((_rootFrame.Content as FrameworkElement)?.DataContext is MainPageViewModel mainPage)
+				{
+					showViewModel = mainPage.Show;
+				}
+
+				if (showViewModel == null) 
```

Comment:
I would just do a single if statement and not check for null, declare a variable, etc.